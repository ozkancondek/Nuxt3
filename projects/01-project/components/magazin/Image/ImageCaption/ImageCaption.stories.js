import ImageComponent from '../Image.vue'
import ImageCaption from './ImageCaption.vue'
import { STORYBOOK_UI_COMPONENTS_PREFIX } from '~/components/storybook.constants'

export default {
  title: `${STORYBOOK_UI_COMPONENTS_PREFIX}/Image/ImageCaption`,
  component: ImageCaption,
  argTypes: {},
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const Template = (args, { argTypes }) => ({
  components: { ImageCaption, ImageComponent },
  props: Object.keys(argTypes),
  template:
    '<div class="tw-max-w-3xl"><ImageCaption v-bind="$props"><ImageComponent v-bind="$props.image.content"/></ImageCaption></div>',
})

export const ImageCaptionExample = Template.bind({})
ImageCaptionExample.args = {
  captionPosition: 'bottom',
  image: {
    type: 'Image',
    content: {
      articleImage: true,
      src: '/api/disc/cms/public/dam/Grill/grillrost-reinigen/grillrost-reinigen.jpg',
      alt: 'Grillrost reinigen',
      caption:
        '<p>Hier wird ein Grillrost gereinigt Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet</p>',
      captionAlignRight: false,
      aspectRatio: 1.78,
      originalAspectRatio: 1.78,
      fpX: 0.5,
      fpY: 0.5,
      fpZoom: 1,
    },
  },
}

export const ImageCaptionInlineExample = Template.bind({})
ImageCaptionInlineExample.args = {
  captionPosition: 'inline',
  image: {
    type: 'Image',
    content: {
      articleImage: true,
      src: '/api/disc/cms/public/dam/Grill/grillrost-reinigen/grillrost-reinigen.jpg',
      alt: 'Grillrost reinigen',
      caption:
        '<p>Hier wird ein Grillrost gereinigt Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet</p>',
      captionAlignRight: false,
      aspectRatio: 1.78,
      originalAspectRatio: 1.78,
      fpX: 0.5,
      fpY: 0.5,
      fpZoom: 1,
    },
  },
}
