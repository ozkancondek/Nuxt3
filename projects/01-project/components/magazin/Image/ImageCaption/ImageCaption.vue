<template>
  <div
    v-if="captionPosition === 'inline'"
    class="tw-flex-row tw-h-full tw-relative"
    data-track-section="image_caption"
  >
    <slot></slot>
    <RichTextParser
      v-if="showCaption"
      class="tw-absolute tw-bottom-0 tw-w-full tw-bg-gray-100 tw-bg-opacity-80 tw-text-sm tw-py-2 tw-px-4"
      :class="image.content.captionAlignRight && 'tw-text-right'"
      :html="image.content.caption"
    ></RichTextParser>
  </div>
  <div v-else class="tw-flex-row" data-track-section="image_caption">
    <slot></slot>
    <RichTextParser
      v-if="showCaption"
      class="tw-py-3 image-caption__caption tw-text-base"
      :class="image.content.captionAlignRight && 'tw-text-right'"
      :html="image.content.caption"
    ></RichTextParser>
  </div>
</template>

<script lang="ts">
import { PropType } from 'vue'
import { Image } from '~/types/cms.types'
import RichTextParser from '~/components/magazin/RichTextParser/RichTextParser.vue'

export default defineComponent({
  name: 'ImageCaption',
  components: { RichTextParser },
  props: {
    image: {
      type: Object as PropType<Image>,
      default: () => undefined,
    },
    captionPosition: {
      type: String as PropType<'bottom' | 'inline'>,
      default: () => 'bottom',
    },
  },
  setup(props) {
    const showCaption = computed(() => {
      return (
        props.image?.content.caption &&
        props.image?.content.caption !== '' &&
        props.image?.content.caption !== '<p></p>' &&
        props.image?.content.caption !== '<p> </p>'
      )
    })
    return { showCaption }
  },
})
</script>

<style scoped>
:where(.disc-image-modal-container) .image-caption__caption {
  padding-left: 12px;
  padding-right: 12px;
}
</style>
