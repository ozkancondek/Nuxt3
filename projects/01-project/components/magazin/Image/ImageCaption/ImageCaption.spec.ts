import { shallowMount } from '@vue/test-utils'
import ImageCaption from '~/components/magazin/Image/ImageCaption/ImageCaption.vue'
export type Image = {
  type: 'Image'
  content: {
    src: string
    alt: string
    caption?: string
    captionAlignRight?: boolean
    aspectRatio: number
    originalAspectRatio?: number
    fpX?: number
    fpY?: number
    fpZoom?: number
    articleImage?: boolean
  }
}


const image: Image = {
  type: 'Image',
  content: {
    articleImage: true,
    src: '/api/disc/cms/public/dam/Grill/grillrost-reinigen/grillrost-reinigen.jpg',
    alt: 'Grillrost reinigen',
    caption:
      '<p>Hier wird ein Grillrost gereinigt Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet</p>',
    captionAlignRight: false,
    aspectRatio: 1.78,
    originalAspectRatio: 1.78,
    fpX: 0.5,
    fpY: 0.5,
    fpZoom: 1,
  },
}
test('renders ImageCaption with bottom caption correctly', () => {
  const wrapper = shallowMount(ImageCaption, {
    mocks: {
      $t: () => {},
    },
    propsData: {
      image,
      captionPosition: 'bottom',
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('renders ImageCaption with inline caption correctly', () => {
  const wrapper = shallowMount(ImageCaption, {
    mocks: {
      $t: () => {},
    },
    propsData: {
      image,
      captionPosition: 'inline',
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})
