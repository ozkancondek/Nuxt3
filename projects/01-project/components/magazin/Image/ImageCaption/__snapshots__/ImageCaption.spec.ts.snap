// Vitest Snapshot v1

exports[`renders ImageCaption with bottom caption correctly 1`] = `
<div
  class="tw-flex-row"
  data-track-section="image_caption"
  data-v-f4f13e04=""
>
  
  
  <rich-text-parser-stub
    class="tw-py-3 image-caption__caption tw-text-base"
    data-v-f4f13e04=""
    html="<p>Hier wird ein Grillrost gereinigt Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet</p>"
  />
</div>
`;

exports[`renders ImageCaption with inline caption correctly 1`] = `
<div
  class="tw-flex-row tw-h-full tw-relative"
  data-track-section="image_caption"
  data-v-f4f13e04=""
>
  
  
  <rich-text-parser-stub
    class="tw-absolute tw-bottom-0 tw-w-full tw-bg-gray-100 tw-bg-opacity-80 tw-text-sm tw-py-2 tw-px-4"
    data-v-f4f13e04=""
    html="<p>Hier wird ein Grillrost gereinigt Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet</p>"
  />
</div>
`;
